tfx/experimental/distributed_inference/graphdef_experiments/subgraph_partitioning/graph_partition.py

```diff
@@ -0,0 +1,439 @@
+"""
+A library for graph partitioning.
+
+The current implementation targets two goals:
+    1. Maximal subgraphs
+    2. Avoid repeated work when running subgraphs in beam
+    
+Definition:
+    1. "op" refers to a graph name. 
+        In our example: op = {'main', 'remote_op_b', 'remote_op_a'}.
+
+    2. "execution_specs" refers to a structure (list) passed to the beam pipeline.
+       We execute spec 1, spec 2, spec 3 in order inside a beam pipeline.
+    
+    <--closer to inputs                   closer to outputs-->
+        --------------------------------------------------
+        |          |          |          |               |
+        |  spec 1  |  spec 2  |  spec 3  | ......        |
+        |          |          |          |               |
+        --------------------------------------------------
+                 /              \
+                /                \
+               / ---------------- \
+                 |   Subgraph   |
+                 ----------------
+                 |    Inputs    |
+                 ----------------
+                 |   Outputs    |
+                 ----------------
+                 | is_remote_op |
+                 ----------------
+                 
+Key Assumptions/Limitations for this implementation:
+    1. Inputs should be GraphDefs.
+    2. Operating on TensorFlow Directed Acyclic Graphs (DAG).
+    3. Only one output per node/op.
+    4. Not supporting tf.functions
+
+@author: jzhunybj
+"""
+
+import tensorflow as tf
+from tensorflow.core.framework import graph_pb2
+
+
+"""Supporting I/O"""
```

Comment:
nit: make this a comment (prefixed by `#`) instead of a docstring or delete it; same below

Reply:
Deleted